### ISSUE TITLE
Network disconnected when using cloud-init and API

### ISSUE THREAD

**Issue by hassenius** [vmware/open-vm-tools#299]
It appears when using cloud-init on linux DeployPkg does not attempt to enable the interfaces
https://gist.github.com/hassenius/4a571955452ad5d1693109a92548f9ee

The problem does not appear when deploying manually from vsphere client, as it seems this will have some default setting to enable NIC at boot. However, several external API driven solutions do not have this as default, or not even configurable.
I've seen this issue mentioned in several different places, including:
[ansible](https://github.com/ansible/ansible/issues/42064)
[terraform](https://github.com/terraform-providers/terraform-provider-vsphere/issues/388)

So, as far as I can see, cloud-init does not have a way to replicate the functionality in: [libDeployPkg/linuxDeployment.c/TryEnableNics](https://github.com/vmware/open-vm-tools/blob/master/open-vm-tools/libDeployPkg/linuxDeployment.c#L811) and this code is never executed in the [CloudInitSetup](https://github.com/vmware/open-vm-tools/blob/master/open-vm-tools/libDeployPkg/linuxDeployment.c#L909) function.

It is also unclear how to ignore cloud init with [VMWAREDEPLOYPKG_HEADER_FLAGS_IGNORE_CLOUD_INIT](https://github.com/vmware/open-vm-tools/blob/master/open-vm-tools/libDeployPkg/linuxDeployment.c#L1184)

In our scenario we just want cloud-init for some simple config injection (ssh keys, etc), we do not want/need cloud-init to manage the network.

With the GOSC workflow raising the interfaces, I don't think it makes sense to force a different usage pattern of the API, rather than making GOSC and cloud-init workflow have the same end result (enabling the interfaces).

**Comment by stanguturi** [comment]
> https://gist.github.com/hassenius/4a571955452ad5d1693109a92548f9ee
Yes. The linuxDeployPkg will not enable the NICS when the 'deployment' task is delegated to cloudinit. 'VMware customization' module in cloud-init enables the NICS. Please check the following 

* https://github.com/cloud-init/cloud-init/blob/master/cloudinit/sources/DataSourceOVF.py (Search for enable_nics)
* https://github.com/cloud-init/cloud-init/blob/master/cloudinit/sources/helpers/vmware/imc/guestcust_util.py (Search for enable_nics)

> In our scenario we just want cloud-init for some simple config injection (ssh keys, etc), we do not want/need cloud-init to manage the network.
Can you please elaborate on your use case?

**Comment by hassenius** [comment]
Thanks @stanguturi I'll dig into these two. We're not using OVF, so I guess I'll look at why the cloud-init guestcust util was not exercised to enabling the nic in any of our various configuration attempts.

For our use case, we're primarily looking to have a single standardised vm template that can:
1. Grow the root partition to any size (was hoping to use the cloud-init growpart module for this)
2. Have no shared credentials, so create users and credentials at creation time (was hoping to use standard cloud-init ways to do this as we do on other platforms)

Since we're already using Terraform for a lot of our automation, we were hoping to create a fairly easy workflow where terraform would create a cdrom with the user data containing the users and keys to create, upload it to a data store and make sure the vm boots with the cdrom mounted.

Since terraform manages the environment it would also clean up the ISO when tearing down the environment.
